subsys/usb/class/dfu/usb_dfu.c

```diff
@@ -556,15 +583,12 @@ static int dfu_class_handle_req(struct usb_setup_packet *pSetup,
 			dfu_data.bytes_sent += len;
 			dfu_data.block_nr++;
 
-			if (dfu_data.bytes_sent == dfu_data.flash_upload_size &&
+			if (dfu_data.bytes_sent == dfu_data.fa->fa_size &&
```

Comment:
flash_upload_size and flash area size are not the same things

Reply:
`flash_upload_size` is set to `fa->fa_size` in the original code (https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/usb/class/usb_dfu.c#L703) and no where else changed; I just move it here to save a bit of space in the `dfu_data` struct. Are there cases when you dont want to access the entire partition during upload?

Reply:
Yes, right